packages/extra/test/reference/DayOfMonthTest.js

```diff
@@ -0,0 +1,600 @@
+/*
+ * @copyright (c) 2022, Philipp Thürwächter & Pattrick Hüper & Michał Sobkiewicz
+ * @copyright (c) 2007-present, Stephen Colebourne & Michael Nascimento Santos
+ * @license BSD-3-Clause (see LICENSE in the root directory of this source tree)
+ */
+
+import {
+    ChronoField,
+    ChronoUnit,
+    Clock,
+    DateTimeException,
+    IsoChronology,
+    IsoFields,
+    LocalDate,
+    LocalTime,
+    Month,
+    MonthDay,
+    NullPointerException,
+    TemporalField,
+    TemporalQueries,
+    UnsupportedTemporalTypeException,
+    ValueRange,
+    YearMonth,
+    ZoneId,
+} from '@js-joda/core';
+
+import '../_init';
+
+import { assertEquals, assertFalse, assertSame, assertThrows } from '../testUtils';
+
+import { DayOfMonth } from '../../src/DayOfMonth';
+
+describe('org.threeten.extra.TestDayOfMonth', () => {
+    const MAX_LENGTH = 31;
+    const TEST = DayOfMonth.of(12);
+
+    class TestingField extends TemporalField {
+        getBaseUnit() {
+            return ChronoUnit.DAYS;
+        }
+
+        getRangeUnit() {
+            return ChronoUnit.MONTHS;
+        }
+
+        range() {
+            return ValueRange.of(1, 28, 31);
+        }
+
+        isDateBased() {
+            return true;
+        }
+
+        isTimeBased() {
+            return false;
+        }
+
+        isSupportedBy(temporal) {
+            return temporal.isSupported(ChronoField.DAY_OF_MONTH);
+        }
+
+        // eslint-disable-next-line no-unused-vars
+        rangeRefinedBy(temporal) {
+            return this.range();
+        }
+
+        getFrom(temporal) {
+            return temporal.getLong(ChronoField.DAY_OF_MONTH);
+        }
+
+        adjustInto(temporal, newValue) {
+            return temporal.with(ChronoField.DAY_OF_MONTH, newValue);
+        }
+    }
+
+    TestingField.INSTANCE = new TestingField();
+
+    //-----------------------------------------------------------------------
+    // now()
+    //-----------------------------------------------------------------------
+    describe('now()', function(/*this*/) {
+        this.retries(100);
```

Comment:
Yes, that's much better than a for loop 👍 . 
An alternative would be to use sinon fakeTimers, might be a bit better readable. 
BTW wouldn't be a retry of one enough? If there is the rare case of a day change between the two calls of now(), the next retry can not have a day change.

Reply:
> An alternative would be to use sinon fakeTimers, might be a bit better readable.

In java version, you would have to mock `Clock`, which doesn't seem to be a good idea (junit itself can be using `Clock` in reports and so on). As this are reference tests, I try to make them as similar to the original as possible - they are going to be easier to maintain.

> BTW wouldn't be a retry of one enough? If there is the rare case of a day change between the two calls of now(), the next retry can not have a day change.

Next retry _should_ not have a day change, but you can't tell for sure. It's clear that more than 100 instead of 2 won't harm, but at the same time I agree that 2 could be more self-descritive. In threeten-extra there is TestOffsetDate.java, where number of retries is 100. I assumed that 100 should be applied consistently in the entire project, but we can wait to see if https://github.com/ThreeTen/threeten-extra/pull/218/commits/41cb65f04d075cba650ae8a01fa96a0360c5033c is going to be merged with same or different number of retries in DayOfMonth tests (if it's going to be merged at all).

Reply:
Indeed, in java it's not recommended to mock time, but in javascript its unproblematic. Anyway, the retry solution is good to go and its a good argument to stick closer to the threeten impl. Nevertheless we should not copy bugs, etc from threeten

>  It's clear that more than 100 instead of 2 won't harm

The problem with 100 is, that in case of an other error, the test get repeated 100 times. Not a real problem, but should only be done if really required.

IMHO `this.retries(1)` is correct and therefore easier to understand. If the first test fails because of a day change, the second attempt will not fail, because the new day will have already started.

Reply:
> IMHO `this.retries(1)` is correct and therefore easier to understand. If the first test fails because of a day change, the second attempt will not fail, because the new day will have already started.

It's still theoretically possible to encounter another day change after first fail (i.e. after NTP sync), so `this.retries(1)` isn't necessarily 100% correct, but I agree that 1 retry may be easier to understand. After all, it's still theoretically possible to encounter 100 fails in a row even if now() was implemented correctly.

I'll change the number of retries to 1 when I get back home.

Reply:
Done. I had to force push because of reword, let me be excused.